test/zipkin/util.go

```diff
@@ -117,29 +135,53 @@ func CheckZipkinPortAvailability() error {
 	return monitoring.CheckPortAvailability(ZipkinPort)
 }
 
-// JSONTrace returns a trace for the given traceId in JSON format
-func JSONTrace(traceID string) (string, error) {
+// JSONTrace returns a trace for the given traceId. It will continually try to get the trace. If the
+// trace it gets has the expected number of spans, then it will be returned. If not, it will try
+// again. If it reaches timeout, then it returns everything it has so far with an error.
+func JSONTrace(traceID string, expected int, timeout time.Duration) ([]model.SpanModel, error) {
```

Comment:
```suggestion
func JSONTrace(traceID string, expected int, timeout time.Duration) (tracesSoFar []model.SpanModel, err error) {
```

Then you can remove all the definitions below.